log4j-core/src/main/java/org/apache/logging/log4j/core/metrics/MetricManager.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.core.metrics;
+
+import java.util.Map;
+import java.util.function.ToDoubleFunction;
+
+import org.apache.logging.log4j.plugins.di.Key;
+
+public interface MetricManager {
```

Comment:
I smell a Dropwizard Metrics, Micrometer, etc. rewrite here. I doubt if such a big undertaking is necessary. If we stick to the listener interface concept I outlined above, maybe we can provide vendor-specific – one for Micrometer, one for Dropwizard Metrics, etc. – modules and avoid rolling out our own implementation.

Reply:
I don't want to rewrite anything; I was developing interfaces that could easily delegate to Micrometer APIs. If we go for an event listener API, then we have to define a bunch of rules about event interpretation which should be logically equivalent to this proposal.

Reply:
I don't see how that's going to be possible without either adding third-party dependencies or implementing facades for all the various OpenTelemetry APIs for internal use.